### PR TITLE
[Feature] Seasonal splash messages logic + scaffolding

### PR DESCRIPTION
**Pull request by flx-sta**
## What are the changes the user will see?

Seasonal splash messages if any are active and it has messages.

## Why am I making these changes?

@damocleas wants seasonal splash messages
https://discord.com/channels/1125469663833370665/1286021055286350006

## What are the changes from a developer perspective?

_See `Files` Tab._
Made sure to add some template types to make _editors_ aware of wrong day-moth formats.

The first (example) batch of messages is:
```json
{
  "halloween": {
    "pumpkaboosAbout": "Pumpkaboos about!",
    "mayContainSpiders": "May contain spiders!",
    "spookyScaryDuskulls": "Spooky, Scary Duskulls!"
  },
  "xmas": {
    "happyHolidays": "Happy Holidays!",
    "delibirdSeason": "Delibird Season!"
  },
  "newYears": {
    "happyNewYear": "Happy New Year!"
  }
}
```

In parallel I've also optimized the way that messages are rendered to reduce unnecessary `i18next.t()` calls and also make the battles-won COUNT variable an actual i18next interpolation

### Screenshots/Videos

image

## How to test the changes?

Make sure one of the prepared season is active.
Add some messages.
Check the console for the console.log that is being printed.
reload a few times hoping your message shows up

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [ ] The PR is self-contained and cannot be split into smaller PRs? => not fully. But I tried my best
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Review by damocleas — APPROVED** [review]

**Review by CodeTappert — DISMISSED** [review]

File: src/locales/ja/splash-messages.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "battlesWon": "Battles Won!",
+  "battlesWon": "{{count, number}} Battles Won!",
```
Comment (Chapybara-jp) [inline src/locales/ja/splash-messages.json:2]:
```suggestion
  "battlesWon": "勝ったバトル：{{count, number}}回！",
```

**Review by damocleas — DISMISSED** [review]

**Review by CodeTappert — DISMISSED** [review]

File: src/locales/es/splash-messages.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "battlesWon": "¡Batallas ganadas!",
+  "battlesWon": "¡{{count}} Batallas ganadas!",
```
Comment (MokaStitcher) [inline src/locales/es/splash-messages.json:2]: no `, number` on this one, is that intended?

**Review by MokaStitcher — DISMISSED** [review]
just one small doubt, otherwise lgtm

File: src/locales/es/splash-messages.json (hunk 2 above)
Comment (flx-sta) [reply src/locales/es/splash-messages.json:2]: nope.. whops
```suggestion
  "battlesWon": "¡{{count, number}} Batallas ganadas!",
```

**Review by CodeTappert — APPROVED** [review]

**Review by damocleas — APPROVED** [review]

**Review by DayKev — APPROVED** [review]

**Pull request merged**